### PR TITLE
add workflow to upload release binaries

### PR DESCRIPTION
**Pull request by Disservin**
The following workflow automatically uploads binaries to a release if a tag was pushed which matches `sf_*`. 

The following action was used for this https://github.com/softprops/action-gh-release

You can find a release/run here https://github.com/Disservin/Stockfish/releases/tag/sf_test_release_v1, https://github.com/Disservin/Stockfish/actions/runs/4725451108

### CODE REVIEW

**Comment by maximmasiutin** [comment]
Could you please make a text file generated inside each archive which command line did you use to build the binary, and the output of the compiler (you can make the script to strip sensitive names like paths, if any). 

Also, please consider using clang (version 16 or later), as using gcc for the official binaries is not justified by ~10% NPS loss.

**Comment by Disservin** [comment]
Yeah I will change that in the next few days ;) I’ll mark it as a draft for now

File: .github/workflows/stockfish_release.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          mkdir stockfish
+          cp -r wiki stockfish/
+          cp -r src stockfish/
+          cp stockfish-$OS-$BINARY$EXT stockfish/
+          cp "Top CPU Contributors.txt" stockfish/
+          cp Copying.txt stockfish/
+          cp AUTHORS stockfish/
+          cp CITATION.cff stockfish/
+          cp README.md stockfish/
+          tar -cvf stockfish-$OS-$BINARY.tar stockfish
```
Comment (UniQP) [inline .github/workflows/stockfish_release.yml:106]: I thought about using `git archive` instead but that doesn't work well with the wiki being in a separate repository.

**Comment by Disservin** [comment]
f****

**Pull request closed without merging**